### DISCUSSION TITLE
Search Engine Postgresql

### DISCUSSION THREAD

**Discussion by sernays** [requarks/wiki#4047] [Error / Bug Report]
**Describe the bug**
Some key-words are not indexed with postgresql, and fact, the search engine have apparently random errors.

For example in a page : the sentence "Nodes du cluster". cluster is indexed, not nodes.
Retry the indexation doesn't work.

Postgresql Logs : 

```
2021-06-03 09:27:05.447 CEST [83953] ERROR: operator does not exist: text <-> unknown at character 19
2021-06-03 09:27:05.447 CEST [83953] HINT: No operator matches the given name and argument types. You might need to add explicit type casts.
2021-06-03 09:27:05.447 CEST [83953] STATEMENT: SELECT word, word <-> $1 AS rank FROM "pagesWords" WHERE similarity(word, $2) > 0.2 ORDER BY rank LIMIT 5;
```

**To Reproduce**

**Screenshots**

**Host Info (please complete the following information):**

OS: Docker
Wiki.js version: 2.5.201
Database engine: postgres 12

**Comment by anli-xsigns** [comment]
Same error here, search is always failing with the same error.

My system:
![Schnappschuss_100221_012400_AM](https://user-images.githubusercontent.com/8227996/135695499-2ae00342-6957-4baa-9489-e8282e02ec38.jpg)

**Reply by joeldeteves** [reply]
I get this error as well, whenever I type something that doesn't exist. Maybe instead of 'unexpected error occurred' it should say 'not found'

**Comment by minamo7sen** [comment]
I had the same error, in my case apparently it was because I enabled the "pg_trgm" extension **before** connecting to the wiki DB.

So obviously, the solution was to connect to the wiki DB first.

> The extension has to be activated per database so make sure to enable it for your Wiki.js database.

Doc. Link: https://docs.requarks.io/search/postgres
